jax/_src/numpy/ufuncs.py

```diff
@@ -817,14 +817,109 @@ def cosh(x: ArrayLike, /) -> Array:
   """
   return lax.cosh(*promote_args_inexact('cosh', x))
 
-@implements(np.arcsinh, module='numpy')
+
 @partial(jit, inline=True)
 def arcsinh(x: ArrayLike, /) -> Array:
+  r"""Calculate element-wise inverse of hyperbolic sine of input.
+
+  JAX implementation of :obj:`numpy.arcsinh`.
+
+  The inverse of hyperbolic sine is defined by:
+
+  .. math::
+
+    arcsinh(x) = \ln(x + \sqrt{1 + x^2})
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array of same shape as ``x`` containing the inverse of hyperbolic sine of
+    each element of ``x``, promoting to inexact dtype.
+
+  Note:
```

Comment:
Should we add a note about the valid domain, similar to `arccosh` and `arctanh`?

Reply:
Thanks! Added a note on valid domain of `arcsinh`.